source/content/guides/composer-convert.md

```diff
@@ -61,7 +61,281 @@ The steps in this process migrate a site, so the new site will no longer maintai
+
+ In the `pantheon.yml` file, the `api_version: 1` and `build_step: true` values are required.
+
+## Add in the Custom and Contrib Code Needed to Run Your Site
+
+What makes your site code unique is your selection of contributed modules and themes, and any custom modules or themes your development team has created. These customizations need to be replicated in your new project structure.
+
+### Contributed Code
+
+The goal of this process is to have Composer manage all the site's contrib modules, contrib themes, core upgrades, and libraries (we'll call this _contributed code_). The only things that should be migrated from the existing site are custom code, custom themes, and custom modules that are specific to the existing site.
+
+#### Modules and Themes
+
+The steps here ensure that any modules and themes from [drupal.org](https://drupal.org) are in the `composer.json` `require` list.
+
+Once Composer is aware of all the contributed code, you'll be able to run `composer update` from within the directory to have Composer upgrade all the contributed code automatically.
+
+Begin by reviewing the existing site's code. Check for contributed modules in `/modules`, `/modules/contrib`, `/sites/all/modules`, and `/sites/all/modules/contrib`.
+
+1. Review the site and make a list of exactly what versions of modules and themes you depend on. One way to do this is to run the `pm:list` Drush command from within a contributed modules folder (e.g. `/modules`, `/themes`, `/themes/contrib`, `/sites/all/themes`, `/sites/all/themes/contrib`, etc.).
+
+  This will list each module followed by the version of that module that is installed:
+
+  ```bash{promptUser:user}
+  terminus drush $SITE.dev pm:list -- --no-core --fields=name,version  --format=table
+  ```
+
+1. You can add these modules to your new codebase using Composer by running the following for each module in the `$SITE` directory:
+
+  ```bash{promptUser:user}
+  composer require drupal/MODULE_NAME:^VERSION
+  ```
+
+<Partial file="module-name.md" />
+
+  Some modules use different version formats.
+
+   - For older-style Drupal version strings:
+
+   ```none
+   Chaos Tools (ctools)  8.x-3.4
+   ```
+
+    Replace the `8.x-` to convert this into `^3.4`
+
+   - Semantic Versioning version strings:
+
+   ```none
+   Devel (devel)  4.1.1
+   ```
+
+    Use the version directly, e.g. `^4.1.1`
+
+    <Partial file="module_name.md" />	  
+
+
+#### Other Composer Packages
+
+If you have added non-Drupal packages to your site via Composer, use the command `composer require` to migrate each package. You can use the following command to display the differences between the master and your current `composer.json`:
+
+```
+git diff master:composer.json composer.json
+```
+
+#### Libraries
+
+Libraries of package-type `drupal-library` can be handled similarly to modules, but the specifics depend on how your library code was included in the source site. 
```

Comment:
we say "the specifics depend on how your library code was included in the source site".  Can we provide any examples of this?